src/u_send_request.c

```diff
@@ -20,8 +20,15 @@
  *
  * You should have received a copy of the GNU General Public
  * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
- * 
+ *
  */
+
+// Enables POSIX functions on non-MSVC targets.
+// This is required for gmtime_r() which is not part of the ISO C standard.
+#ifndef _MSC_VER
+  #define _POSIX_C_SOURCE 200112L
+#endif
```

Comment:
As you mentionned in your issue, this is specific to MinGW-w64, so I'd prefer to add another `#if` to specify it, otherwise the `#define _POSIX_C_SOURCE 200112L` will apply to all non MSC compilers, like gcc or clang on linux.

Reply:
This enables compilation with MinGW-w64 yes, but in `u_send_request.c` you're using function `gmtime_r()` which is part of the POSIX standard but not the ISO C standard so withoud that `#define` it should not compile with any standard compliant compiler.

Reply:
I mean, this should work with both gcc and clang and it explicitly states that some POSIX extensions are used in that file.

Reply:
https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html

Reply:
If you don't like it it can be changed to something like `#if defined (__MINGW32__) || defined (__MINGW64__)` but I still believe that this is the correct way to do it because current version depends on default value of `-std=` and that can be quite different on different versions of gcc and different platforms.

Reply:
You're probably right, let me make some tests and if so, I'll merge your pull request soon

Reply:
Even if I couldn't find any problem with it on my side, I'm still not confident that it wouldn't break another platform build, so I suggest the following code:

```C
#if !defined(_MSC_VER) && !defined(_POSIX_C_SOURCE)
  #define _POSIX_C_SOURCE 200112L
#endif
```

What do you think?